api/src/pcapi/scripts/offer/migrate_support_physique_musique_subcategory.py

```diff
@@ -0,0 +1,91 @@
+import logging
+from typing import Iterator
+
+from pcapi.core.categories.subcategories_v2 import SUPPORT_PHYSIQUE_MUSIQUE
+from pcapi.core.categories.subcategories_v2 import SUPPORT_PHYSIQUE_MUSIQUE_CD
+from pcapi.core.offers.models import Offer
+from pcapi.core.offers.models import Product
+from pcapi.models import db
+from pcapi.repository import transaction
+
+
+logger = logging.getLogger(__name__)
+
+
+CHUNK_SIZE = 1000
+
+
+def migrate_support_physique_musique_products(from_id: int | None = None) -> None:
+    product_pages_to_update = get_music_product_pages(from_id)
+    for product_page in product_pages_to_update:
+        with transaction():
+            for product in product_page:
+                product.subcategoryId = SUPPORT_PHYSIQUE_MUSIQUE_CD.id
+            db.session.add_all(product_page)
+
+
+def get_music_product_pages(from_id: int | None = None) -> Iterator[list[Product]]:
+    products_query_base = Product.query.filter(Product.subcategoryId == SUPPORT_PHYSIQUE_MUSIQUE.id).order_by(
+        Product.id
+    )
+    products_query = products_query_base
+    if from_id is not None:
+        products_query = products_query.filter(Product.id > from_id)
+
+    total_count = products_query.count()
+
+    product_page_size = None
+    counter = 0
+    while product_page_size is None or product_page_size == CHUNK_SIZE:
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏J'ai du mal à comprendre ton while ici..

Tu as `total_count` et `counter` juste au dessus, pourquoi tu fais pas un 
```python 
count = 0
while count < total_count:
    [...]

Reply:
dommage que mettre le chiffre en dur ne soit pas une option pour les offres